tvb/recon/flow/parc_fod_to_connectome.py

```diff
@@ -0,0 +1,153 @@
+
+        # Normalize track counts
+        ntracks = np.sum(track_count_mtx)
+        track_count_mtx /= ntracks
+
+        return mean_len_mtx, track_count_mtx
+
+    def _gen_connectome(self, runner: Runner, tracks: os.PathLike, ntracks: int, parc_lbl: os.PathLike) \
+            -> (np.ndarray, np.ndarray, np.ndarray):
+        """Generate a connectome and track statistics from tracks file."""
+
+        # Sift the tracks
+        ntracks_sift = int(ntracks * self.FRACTION_SIFT)
+        tracks_sifted = runner.tmp_fname('tracks_sifted_N%s.tck' % ntracks)
+        runner.run(mrtrix.run_tcksift(tracks, self.fod, tracks_sifted, ntracks_sift))
+
+        # Save the lengths of all tracks
+        track_lengths = runner.tmp_fname('track_lengths_N%s.txt' % ntracks)
+        runner.run(mrtrix.dump_streamlines_length(tracks_sifted, track_lengths))
+
+        # Generate the connnectome and save the track assignments
+        conn = runner.tmp_fname('conn_N%s.csv' % ntracks)
+        track_assignments = runner.tmp_fname('track_assignments_N%s.txt' % ntracks)
+        runner.run(mrtrix.run_tck2connectome(tracks_sifted, parc_lbl, conn,
+                                             assignment=mrtrix.tck2connectome.Assignment.radial_search(2.0),
+                                             out_assignments=track_assignments))
+
+        # Get the track statistics
+        mean_len_mtx, track_count_mtx = self._get_track_stats(track_assignments, track_lengths)
+
+        return mean_len_mtx, track_count_mtx
+
+    def run(self, runner: Runner):
+        log = logging.getLogger('parc_fod_to_connectome')
+        tic = time.time()
+
+        log.info('Relabeling parcellation')
+        lut_in = File(os.path.join(os.environ['FREESURFER_HOME'], 'FreeSurferColorLUT.txt'))
+        lut_out = File(os.path.join(os.environ['MRT3'], 'src/connectome/tables/fs_default.txt'))
+        parc_lbl = runner.tmp_fname('parc_lbl.nii.gz')
+        runner.run(mrtrix.run_labelconvert(self.parc, lut_in, lut_out, parc_lbl))
+
+        ntracks = self.MIN_TRACKS
+        conv_history = []
+
+        # Generate initial number of tracks, the connectome and its statistics
+        log.info('Generating initial %i tracks' % ntracks)
+        tracks_a = runner.tmp_fname('tracks_A_N%s.tck' % ntracks)
+        runner.run(mrtrix.run_tckgen(self.fod, tracks_a, ntracks, seed_gmwmi=self.gmwmi, act=self.ftt))
+
+        log.info('Generating the connectome from %i tracks' % ntracks)
+        mean_len_mtx_0, track_count_mtx_0 = self._gen_connectome(runner, tracks_a, ntracks, parc_lbl)
+
+        converged = False
+        while ntracks <= self.MAX_TRACKS/2:
+            # TODO: catching errors, warnings
+
+            log.info('Generating additional %i tracks' % ntracks)
+            tracks_b = runner.tmp_fname('tracks_B_N%s.tck' % ntracks)
+            runner.run(mrtrix.run_tckgen(self.fod, tracks_b, ntracks, seed_gmwmi=self.gmwmi, act=self.ftt))
```

Comment:
I hadn't thought enough about this.  This indirection of using a `runner` isn't sophisticated enough to describe things like convergence.

I am starting to doubt the wisdom of this idea.  It would be drastically simpler to just run the commands directly.